### ISSUE TITLE
ZeroDivisionError during PonAbe liftover

### ISSUE THREAD

**Issue by grahamgower** [popsim-consortium/stdpopsim#1249]
Get necessary files.
```
$ wget https://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/liftOver
$ wget https://hgdownload.soe.ucsc.edu/goldenPath/ponAbe3/liftOver/ponAbe3ToPonAbe2.over.chain.gz
$ wget https://hgdownload.soe.ucsc.edu/goldenPath/ponAbe2/liftOver/ponAbe2ToPonAbe3.over.chain.gz
```
Be sure to `chmod +x liftOver` and put it in your `$PATH`. Then try the `liftOver_catalog.py` script as outlined in the dev docs.

```
$ python maintenance/liftOver_catalog.py \
  --species PonAbe \
  --map NaterPA_PonAbe2 \
  --chainFile ponAbe2ToPonAbe3.over.chain.gz \
  --validationChain ponAbe3ToPonAbe2.over.chain.gz \
  --winLen 1000 \
  --gapThresh 1000000 \
  --useAdjacentAvg \
  --retainIntermediates

Downloading tarball...
Extracting tarball...

Mapping the rates to the new assembly...
Reading liftover chains
Mapping coordinates
Reading liftover chains
Reading liftover chains
Mapping coordinates
Mapping coordinates
Reading liftover chains
Reading liftover chains
Mapping coordinates
Mapping coordinates
Reading liftover chains
Mapping coordinates
Reading liftover chains
Reading liftover chains
Mapping coordinates
Mapping coordinates
Process Process-2:
Traceback (most recent call last):
  File "/usr/lib/python3.10/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.10/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/grg/src/stdpopsim/maintenance/liftOver.py", line 561, in worker
    joinedRates, new_average = bed_to_hapmap(
  File "/home/grg/src/stdpopsim/maintenance/liftOver.py", line 143, in bed_to_hapmap
    rescaled_rate = rescale_rate(oldRateDict, ar)
  File "/home/grg/src/stdpopsim/maintenance/liftOver.py", line 124, in rescale_rate
    float(ar[3]) * (old_interval[1] - old_interval[0]) / (int(ar[2]) - int(ar[1]))
ZeroDivisionError: float division by zero
```

**Comment by grahamgower** [comment]
Possibly related to #595.

**Comment by grahamgower** [comment]
`liftOver` is outputting 0-length bed intervals.
```
awk '$2==$3' genetic_map_PonAbe3_1.txt.tmp2 |head
chr1    344952  344952  0.261933        int2723
chr1    963630  963630  0.004067        int6586
chr1    2098258 2098258 11.502701       int13362
chr1    4132776 4132776 1.317146        int28819
chr1    4136651 4136651 0.373152        int28849
chr1    4138772 4138772 0.337640        int28865
chr1    4157123 4157123 0.636030        int28996
chr1    4176743 4176743 3.570152        int29126
chr1    4192872 4192872 0.731573        int29296
chr1    8095452 8095452 0.019146        int60004
```

**Comment by grahamgower** [comment]
The `hapmap_to_bed()` function in `maintenance/liftOver.py` is producing 0-length intervals, which are then given to `liftOver`. I guess there's an off-by-one error here, but the code is hard to follow.

**Comment by grahamgower** [comment]
It's a problem with the recombination map, which itself contains 0-length intervals. How is one to interpret this (3rd column is the rate (cM/Mb), the 4th column is the genetic position (cM))?
```
awk '$2==524260' ~/.cache/stdpopsim/genetic_maps/PonAbe/NaterPA_PonAbe2/Nater_et_al_PA_chr1_PonAbe2.txt
chr1    524260  0.261933        0.246834
chr1    524260  0.322591        0.246834
```

**Comment by petrelharp** [comment]
Um wow that's weird. I think we just have to take it at face value, though, and asusme that that recombination rate applies to zero base pairs? (It's probably a bug generating the map somehow, but we're not trying to debug that?)
